jwst/cube_build/blot_cube_build.py

```diff
@@ -324,47 +285,51 @@ def blot_images_nirspec(self):
 
                 # median cube ra,dec,wave -> x_slice, y_slice
                 x_slice, y_slice = slice_wcs.invert(ra_use, dec_use, wave_use)
-
                 x_slice = np.ndarray.flatten(x_slice)
                 y_slice = np.ndarray.flatten(y_slice)
                 flux_slice = np.ndarray.flatten(flux_use)
 
+                # only use values what fall in bounding box of the slice
                 xlimit, ylimit = slice_wcs.bounding_box
                 xuse = np.logical_and(x_slice >= xlimit[0], x_slice <= xlimit[1])
                 yuse = np.logical_and(y_slice >= ylimit[0], y_slice <= ylimit[1])
+                fgood = (flux_slice != 0)  # edge cube pixel may = 0
 
-                fuse = np.logical_and(xuse, yuse)
+                # fuse = np.logical_and(xuse, yuse)
+                fuse = np.where(fgood & xuse & yuse)
                 x_slice = x_slice[fuse]
                 y_slice = y_slice[fuse]
                 flux_slice = flux_slice[fuse]
 
-                nn = flux_slice.size
-                for ipt in range(nn):
-                    # search xcenter and ycenter seperately. These arrays are smallish.
-                    # xcenter size = naxis1 on detector
-                    # ycenter size = naxis2 on detector
-                    xdistance = np.absolute(x_slice[ipt] - xcenter)
-                    ydistance = np.absolute(y_slice[ipt] - ycenter)
-
-                    index_x = np.where(xdistance <= roi_det)
-                    index_y = np.where(ydistance <= roi_det)
-
-                    if len(index_x[0]) > 0 and len(index_y[0]) > 0:
-                        d1pix = np.array(x_slice[ipt] - xcenter[index_x])
-                        d2pix = np.array(y_slice[ipt] - ycenter[index_y])
-
-                        dxy = [(dx * dx + dy * dy) for dy in d2pix for dx in d1pix]
-                        dxy = np.sqrt(dxy)
-                        weight_distance = np.exp(-dxy)
-                        weighted_flux = weight_distance * flux_slice[ipt]
-                        index2d = [iy * blot_xsize + ix for iy in index_y[0] for ix in (index_x[0])]
-                        blot_flux[index2d] = blot_flux[index2d] + weighted_flux
-                        blot_weight[index2d] = blot_weight[index2d] + weight_distance
-                ts1 = time.time()
-                log.debug(f"Time to map 1 slice  =  {ts1-ts0:.1f}")
+                nslice = len(x_slice)
+                for i in range(nslice):
+                    x_total.append(x_slice[i])
+                    y_total.append(y_slice[i])
+                    flux_total.append(flux_slice[i])
```

Comment:
Not sure what this loop is doing, but it seems like it could be replaced with something like this:
```suggestion
                x_total.extend(x_slice)
                y_total.extend(y_slice)
                flux_total.extend(flux_slice)
```

Reply:
I originally had what you suggested but then x_total was a list of arrays instead of one continuous array. I could not figure out how to turn the list of arrays into a single continuous array ?

Reply:
I have no idea how appending numpy arrays to a Python list could produce "continuous arrays". I do not know what you did previously but I get identical results using both methods. I do not see how the two appraches could produce different results. If you want "continuous arrays" (not sure what that means: a large numpy array holding all smaller arrays stacked?) - then neither code is optimal.